### DISCUSSION TITLE
Questions about iOS

### DISCUSSION THREAD

**Discussion by sunWaterMood** [strongswan/strongswan#1161] [General]
Hi, can you provide a demo for iOS? I'm currently having some issues, currently using the p12 certificate validation keeps failing

**Comment by tobiasbrunner** [comment]
What do you mean with "demo"? Anyway, please refer to our docs for info regarding [Apple clients](https://docs.strongswan.org/docs/5.9/interop/ios.html).

**Comment by gitbls** [comment]
iOS has a strongSwan-compatible client built in.

If you want a quick and easy strongSwan set up on a Debian-like server system to try out connecting to it, have a look at https://github.com/gitbls/pistrong. The installer will install all the necessary packages. There is a script to build the CA, and then you can make Certs that you install on iOS. It's all well-documented.

Once you have both ends installed, it will just work because pistrong eliminates all the Cert confusion.
